src/core/website_manager.py

```diff
@@ -1,311 +1,218 @@
-                cache_dir=False
-            )
-        except (ConnectionError, SuffixListNotFound) as e:
-            self._logger.error(
-                f"Cannot extract top_level_domain because '{e.args}'"
-            )
-            self.tld_extractor = None
-
-        self.reset()
-
-    def load_website_data(self, message: dict = None) -> None:
-        if message is None:
-            return
-
-        if self.website_data.url == "":
-            self.website_data.url = message[MESSAGE_URL]
-            self._extract_top_level_domain()
-
-        if message[MESSAGE_HTML] == "":
-            response = self._get_html_and_har(self.website_data.url)
-            message[MESSAGE_HTML] = response[MESSAGE_HTML]
-            message[MESSAGE_HAR] = response[MESSAGE_HAR]
-            message[MESSAGE_HEADERS] = response[MESSAGE_HEADERS]
-
-        if message[MESSAGE_HEADERS] != "":
-            self.website_data.raw_header = message[MESSAGE_HEADERS]
-            self._preprocess_header()
-
-        if message[MESSAGE_HTML] != "":
-            self.website_data.html = message[MESSAGE_HTML]
-            self._create_html_soup()
-            self._extract_images()
-            self._extract_raw_links()
-            self._extract_extensions()
-
-        if message[MESSAGE_HAR] != "":
-            self._load_har(message[MESSAGE_HAR])
-
-    def _extract_top_level_domain(self) -> None:
-        try:
-            if self.tld_extractor is not None:
-                host = self.tld_extractor(
-                    url=self.website_data.url.replace("http://", "").replace(
-                        "https://", ""
-                    )
-                )
-                domain = [host.subdomain, host.domain, host.suffix]
-                self.website_data.domain = ".".join(
-                    [element for element in domain if element != ""]
-                )
-        except (ConnectionError, SuffixListNotFound, ValueError) as e:
-            self._logger.error(
-                f"Cannot extract top_level_domain because '{e.args}'"
-            )
-            self.website_data.domain = ""
-
-    def _preprocess_header(self) -> None:
-        header: str = self.website_data.raw_header.lower()
-
-        idx = max(header.find('b"'), header.find("b'"))
-        if idx >= 0:
-            header = (
-                header.replace("b'", '"')
-                .replace('b"', '"')
-                .replace("/'", '"')
-                .replace("'", '"')
-                .replace('""', '"')
-                .replace('/"', "/")
-            )
+    url: str
+    html: str
+    raw_header: str
+    soup: BeautifulSoup
+    har: dict[str, Any]
+    domain: str
+    """A dictionary with the http headers where all keys have been converted to lowercase."""
+    headers: dict[str, str]
+    values: list[str]  # fixme: remove
+    raw_links: list[str]
+    image_links: list[str]
+    extensions: list[str]
+
+    @classmethod
+    def fetch_content(
```

Comment:
This class could basically be cut out of the WebsiteManager, right? It does not need to reside here - and is being called once only, anyway.